### ISSUE TITLE
chore: update tech stack

### ISSUE THREAD

**Issue by leog** [fluidemocracy/frontend-legacy#1]
This is the old version of the app using Lua, we want to be able to convert it to something more maintainable by the community, such as next.js with prisma, nuxt, etc.

Take a look at https://github.com/fluidemocracy/backend-legacy which provides the stored procedures for this frontend to work. Instructions to install locally are in the INSTALL/README file for each repository.

**Comment by rishi-raj-jain** [comment]
Hey there,
I'm free to use the stack, right?

**Comment by leog** [comment]
> Hey there,
> I'm free to use the stack, right?

Hi @rishi-raj-jain.
I added you to the team that has access to the repo with the scaffolded app to kickstart development here: https://github.com/fluidemocracy/app. Next.js with Prisma mainly. Hooked up Vercel as well.

> I'll setup both backend and frontend legacies and see what's going on in the whole app.
Great!

> I can then prototype the frontend same in Next.js i believe.
Right, take a look at the app repo based on t3-complete, which uses Next.js with shadcn/ui

Just to clear it out, this is specifically for the frontend, right?
Backend legacy is needed to make frontend work, so porting that would be needed as well, hence why the bounty is high. But should be fairly straightforward, all based on stored procedures.

**Comment by rishi-raj-jain** [comment]
> > Hey there,
> > I'm free to use the stack, right?
> 
> Hi @rishi-raj-jain. I added you to the team that has access to the repo with the scaffolded app to kickstart development here: https://github.com/fluidemocracy/app. Next.js with Prisma mainly. Hooked up Vercel as well.
> 
> > I'll setup both backend and frontend legacies and see what's going on in the whole app.
> 
> Great!
> 
> > I can then prototype the frontend same in Next.js i believe.
> 
> Right, take a look at the app repo based on t3-complete, which uses Next.js with shadcn/ui
> 
> > Just to clear it out, this is specifically for the frontend, right?
> 
> Backend legacy is needed to make frontend work, so porting that would be needed as well, hence why the bounty is high. But should be fairly straightforward, all based on stored procedures.

Sounds good, I'll be on it and shall complete by tomorrow hopefully. Will do my quickest best.

**Comment by leog** [comment]
> Sounds good, I'll be on it and shall complete by tomorrow hopefully. Will do my quickest best.

@rishi-raj-jain  Please take as much time as you need, no specific due date exist, it's better to take the necessary time to do it well.

**Comment by yashug** [comment]
its official now. you can easily call yourself as bounty hunter @rishi-raj-jain. looks like not even sleeping properly to get the first attempt.

Jokes apart. i would like to work on this as well. 

@rishi-raj-jain we can work collaboratively if interested but not a mandatory, if you already on edge to finish this ASAP. let me know. let me know

**Comment by Roshan-Horo** [comment]
I also want to work on this, please let me know @rishi-raj-jain.

**Comment by rishi-raj-jain** [comment]
@yashug & @Roshan-Horo feel free to collaborate. I'm unable to purse this.

**Comment by yashug** [comment]
@leog can this be possible to do easily, if one is not aware of Lua  ?

**Comment by leog** [comment]
> can this be possible to do easily, if one is not aware of Lua ?

@yashug I don't think it could be possible, knowing Lua is essential, which is the second reason the bounty is high

**Comment by rishi-raj-jain** [comment]
@leog 

I am on this again. Do you have the lua app deployed somewhere where I can take a look at the current state of the frontend?

**Comment by leog** [comment]
> I am on this again. Do you have the lua app deployed somewhere where I can take a look at the current state of the frontend (that is in Lua)?

@rishi-raj-jain I have a dockerized version of the Lua app [here](https://github.com/leog/liquidfeedbackv4), not working 100% yet unfortunately, there's some combination of configurations that I didn't get right yet, you could experiment a bit. Between the exposed port and the `myconfig.lua` file and properties `config.absolute_base_url` and `config.localhost`.

The best I could get it to work is by using the following values exposing 8080 port as `make run` currently does:
```lua
config.absolute_base_url = "http://localhost:8080/lf/"
config.localhost = false
```

**Comment by rishi-raj-jain** [comment]
Hey, would it be possible to get a running demo of the app first? The way I was going to suggest next is to split the bounty into two.

- Frontend ($X) -> That setups and uses the latest frameworks/app like Next.js, SvelteKit, etc.
- Backend ($500-$X) -> That setups and uses the latest tech maybe like neon, prisma, etc.

This allows people get paid easier + you to get your issues out quickly.

Would love to work on the frontend / backend, once it's split... Considering that just starting backend is now going to take effort (because some of the errors as below), I think it also deserves a granular bounty.

<img width="772" alt="Screenshot 2023-08-29 at 11 25 35 AM" src="https://github.com/fluidemocracy/frontend-legacy/assets/46300090/4314dde6-8cfa-4299-aa25-ee57bb3fcccb">

Hope this makes sense.

**Comment by leog** [comment]
@rishi-raj-jain got it working in the latest https://github.com/leog/liquidfeedbackv4, you shouldn't get any error, frontend and backend work very much together in this app, as the "backend" is just a collection of stored procedures that are called throughout the frontend app to make it work

Also, luckily, there is a trace option turned on by default in the shared dockerized version of the legacy app where you can peek what is going on behind the scenes, section always available at the bottom of the loaded page, which should help move quicker. See below:

<kbd><img src="https://github.com/fluidemocracy/frontend-legacy/assets/467258/c7e00d9d-122c-4392-bfb5-c20ce0c7ed76" /></kbd>

To create an admin user in the dockerized app, use the invite code I left available "sesam".

Having said that, I do agree the effort to convert the stored procedures into Next.js API functions or tRPC handlers has its own effort to recognize, so I added its own bounty here: https://github.com/fluidemocracy/backend-legacy/issues/1.

**Comment by rishi-raj-jain** [comment]
Hey @leog,
I was able to run this! Thanks for helping me with that.

<img width="1440" alt="Screenshot 2023-09-12 at 9 55 28 AM" src="https://github.com/fluidemocracy/frontend-legacy/assets/46300090/80ea52e8-5416-417f-a936-a37ac53fd598">

How about I build the same thing in JS with https://ui.shadcn.com and some changes?

**Comment by rishi-raj-jain** [comment]
I've asked @zcesur to put my attempt back on!

**Comment by rishi-raj-jain** [comment]
> How about I build the same thing in JS with https://ui.shadcn.com/ and some changes?

Ah I see, you're already using it in /app.

**Comment by rishi-raj-jain** [comment]
@leog 

Can we have a quick call for this?

**Comment by rishi-raj-jain** [comment]
@leog

Can we have a quick call for this?

**Comment by leog** [comment]
@rishi-raj-jain sure, book here https://cal.com/leog

**Comment by rishi-raj-jain** [comment]
@leog 

I've just booked it.
